examples/pizza/web/index.js

```diff
@@ -0,0 +1,144 @@
+var crustSelectEl = document.getElementById('crust-type');
+var toppingsEls = document.querySelectorAll('[name=toppings]');
+var ovenTempEl = document.getElementById('oven-temperature');
+var crustTypeEl = document.getElementById('crust-type');
+var outputEl = document.getElementById('output');
+
+// ¯\_(ツ)_/¯
+function swap16(val) {
+  // le to be
+  return ((val & 0xFF) << 8)
+    | ((val >> 8) & 0xFF);
+}
+const PizzaCrust = vals.PizzaCrust;
+const PizzaToppings = vals.PizzaToppings;
+const PizzaBakeResult = vals.PizzaBakeResult;
+
+var cachedCharacteristics = {};
+
+// current bluetooth connection obj
+var ovenServer = null;
+
+// connect to bluetooth peripheral
+var readyOven = function() {
+  return navigator.bluetooth.requestDevice({
+    filters: [{services: serviceUuids}, {name: name}]
+  }).then(function(device) {
+    return device.gatt.connect();
+
+  }).then(function(server) {
+    ovenServer = server;
+    return server.getPrimaryService(serviceUuids[0]);
+
+  }).then(function(service) {
+    return Promise.all(characteristicUuids.map((characteristic)=>service.getCharacteristic(characteristic)));
+
+  }).then(function(characteristics) {
+    characteristicNames.forEach(function(name, i) {
+      cachedCharacteristics[name] = characteristics[i];
+    });
+    return cachedCharacteristics;
+
+  }).catch(function(err) {
+    alert('oven (bluetooth) error');
+    throw err;
+  });
+};
+
+// characteristic setup
+var readyCrust = function(crustType) {
+  console.log('crustType:', crustType);
+  return function() {
+    var crust = new Uint8Array(1);
+    crust[0] = crustType;
+
+    var pizzaCrustCharacteristic = cachedCharacteristics['crust'];
+    if(pizzaCrustCharacteristic == null) throw new Error('oven not ready!');
+    return pizzaCrustCharacteristic.writeValue(crust).catch(function(err) {
+      alert('crust error');
+      throw err;
+    });
+  }
+};
+
+var readyToppings = function(toppings) {
+  console.log('toppings:', toppings);
+  return function() {
```

Comment:
When I had it in that `reduce` (which you pointed out earlier) I needed the closure.